### PR TITLE
use asyncio.gather

### PR DESCRIPTION
**Pull request by wochinge**
**Proposed changes**:
- use `asyncio.gather` for coroutines which are independent

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa_nlu#code-style) for instructions)

### CODE REVIEW

File: rasa/core/train.py [hunk 1]
```diff
@@ -107,7 +108,7 @@
                 )
 
                 with TempDirectoryPath(tempfile.mkdtemp()) as train_path:
-                    await train(
+                    train_coroutine = train(
```
Comment (erohmensing) [inline rasa/core/train.py]: Is this the sort of typing you always get at me for? 😄 (i forget the name)

File: rasa/train.py [hunk 2]
```diff
@@ -134,10 +134,11 @@ def train(
     Returns:
         Path of the trained model archive.
     """
-    new_fingerprint = await model.model_fingerprint(file_importer)
-
-    stories = await file_importer.get_stories()
-    nlu_data = await file_importer.get_nlu_data()
```
Comment (erohmensing) [inline rasa/train.py]: How was the fingerprint created before getting the stories and NLU before? Doesn't the fingerprint depend on the stories and such?

Comment (wochinge) [reply rasa/train.py]: just depends on the file importer. Moved it back down

File: rasa/core/train.py (hunk 1 above)
Comment (wochinge) [reply rasa/core/train.py]: hungarian notation 😀You're completely right. I assigned a variable to not chain too many calls in `asyncio.gather`. But it's actually not to bad, moved it back in.

File: rasa/train.py [hunk 3]
```diff
@@ -135,8 +135,12 @@ def train(
     Returns:
         Path of the trained model archive.
     """
-    stories = await file_importer.get_stories()
-    nlu_data = await file_importer.get_nlu_data()
+
+    stories, nlu_data = await asyncio.gather(
```
Comment (erohmensing) [inline rasa/train.py]: I think this needs `_, stories, nlu_data`

Comment (wochinge) [reply rasa/train.py]: the fingerprint should not have been in there. Good catch :+1:

Comment (erohmensing) [reply rasa/train.py]: wrong catch, but good nonetheless 😉

**Pull request merged**